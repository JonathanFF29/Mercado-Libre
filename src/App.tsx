import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { ThemeProvider } from 'styled-components';
import { GlobalStyle, theme } from './assets';
import ProductContainer from './container/productContainer';
import rootReducer from './redux/reducer/root-reducer';
import { Provider } from 'react-redux';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)))
function App() {
  return (
    <Provider store={store}>

      <ThemeProvider theme={theme}>
        <GlobalStyle />
        <BrowserRouter>
          <Route path='/'
            render={props => <ProductContainer {...props} />} 
          />
        </BrowserRouter>
      </ThemeProvider>
    </Provider>
  );
}

export default App;

