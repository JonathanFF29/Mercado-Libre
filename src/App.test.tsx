import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import App from './App';

configure({adapter: new Adapter()})

describe('<SearchHeader>', () => {
    it('should should render app component' , () => {
      const component = shallow(<App />);
      expect(component).toMatchSnapshot();
    })
})