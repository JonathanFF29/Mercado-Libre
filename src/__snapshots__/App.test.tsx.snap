// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchHeader> should should render app component 1`] = `ShallowWrapper {}`;

exports[`renders the component 1`] = `ShallowWrapper {}`;
